File: src/main/java/de/pbauerochse/worklogviewer/youtrack/v20174/types/WorkItemBasedGrouping.java
package de.pbauerochse.worklogviewer.youtrack.v20174.types;

import com.fasterxml.jackson.annotation.*;
import de.pbauerochse.worklogviewer.youtrack.domain.GroupByCategory;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
public class WorkItemBasedGrouping implements Grouping, GroupByCategory {

    private final GroupingField field;

    @JsonCreator
    public WorkItemBasedGrouping(@JsonProperty("field") GroupingField field) {
        this.field = field;
    }

    @Override
    public String getId() {
        return field.getId();
    }

    @Override
    public String getPresentation() {
        return field.getPresentation();
    }

    @Override
    public GroupingField getField() {
        return field;
    }

    @JsonIgnore
    @Override
    public String getName() {
        return getPresentation();
    }

    @JsonIgnore
    @Override
    public boolean isValidYouTrackCategory() {
        return true;
    }
}

File: connectors/v20182/src/main/java/de/pbauerochse/worklogviewer/connector/v2018/domain/report/WorkItemBasedGrouping.kt
package de.pbauerochse.worklogviewer.connector.v2018.domain.report

import com.fasterxml.jackson.annotation.*
import de.pbauerochse.worklogviewer.connector.v2018.domain.groupby.GroupingField

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
class WorkItemBasedGrouping @JsonCreator constructor(
    @JsonProperty("field") val field: GroupingField
) : Grouping {

    @JsonIgnore
    override val id: String = field.id

    @JsonIgnore
    override fun getLabel(): String = field.presentation
}